File: backend/models/memory.model.js
const mongoose = require("mongoose");

const memorySchema = new mongoose.Schema(
  {
    childId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Child",
      required: true,
    },
    image: {
      type: String,
      default: `https://${process.env.AWS_S3_BUCKET}.s3.${process.env.AWS_REGION}.amazonaws.com/uploads/memory-default.jpg`, // تعديل الـ default
      validate: {
        validator: function (value) {
          return !value || /\.(jpg|jpeg|png|gif)$/i.test(value);
        },
        message: "Image must be a valid image file",
      },
    },
    description: {
      type: String,
      trim: true,
      maxlength: [500, "Notes cannot exceed 500 characters"],
    },
    date: {
      type: Date,
      default: Date.now,
    },
    time: {
      type: String,
      required: true,
    },
    isFavorite: {
      type: Boolean,
      default: false,
    },
  },
  { timestamps: true }
);

const Memory = mongoose.model("Memory", memorySchema);
module.exports = Memory;

File: Sigma_App/lib/models/memory_model.dart
class Memory {
  final String id;
  final String image;
  final String description;
  final DateTime date;
  final String time;
  final bool isFavorite;

  Memory({
    required this.id,
    required this.image,
    required this.description,
    required this.date,
    required this.time,
    required this.isFavorite,
  });

  factory Memory.fromJson(Map<String, dynamic> json) {
    print('Parsing JSON: $json'); // تتبع القيم اللي بتيجي
    return Memory(
      id: json['_id'] as String? ?? '',
      image: json['image'] as String? ?? '',
      description: json['description'] as String? ?? '',
      date: (json['date'] != null)
          ? (DateTime.tryParse(json['date'] as String)?.toLocal() ?? DateTime.now())
          : DateTime.now(),
      time: json['time'] as String? ?? '00:00',
      isFavorite: json['isFavorite'] as bool? ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'image': image.isEmpty ? null : image,
      'description': description,
      'date': date.toIso8601String(),
      'time': time,
      'isFavorite': isFavorite,
    };
  }

  Memory copyWith({
    String? id,
    String? image,
    String? description,
    DateTime? date,
    String? time,
    bool? isFavorite,
  }) {
    return Memory(
      id: id ?? this.id,
      image: image ?? this.image,
      description: description ?? this.description,
      date: date ?? this.date,
      time: time ?? this.time,
      isFavorite: isFavorite ?? this.isFavorite,
    );
  }
}